klippa/src/main.rs

```diff
@@ -0,0 +1,104 @@
+//! binary subset tool
+//!
+//! Takes a font file and a subset input which describes the desired subset, and ouput is a new
+//! font file containing only the data specified in the input.
+//!
+
+use font_types::GlyphId;
+use klippa::{Plan, Subset};
+use read_fonts::{FontRef, TableProvider, TopLevelTable};
+use std::collections::BTreeSet;
+use thiserror::Error;
+use write_fonts::{
+    from_obj::FromTableRef, tables::hhea::Hhea, tables::hmtx::Hmtx, tables::maxp::Maxp, FontBuilder,
+};
+
+#[derive(Error, Debug)]
+pub enum InvalidInputError {
+    #[error("Invalid input gid {0}")]
+    InvalidGidError(String),
```

Comment:
no need to convert the gid to a string, just keep it as a u16 or a `GlyphId` or whatever it was, you can still print it.


Also we can just call this `InvalidGid`, the 'Error' bit is redundant I think.

Reply:
oh I have a question: this error is emitted when parsing a `&str` to `u16`, so I assume at this time we only have `&str` available?  Can I just change `String` to `&str`?

Reply:
oh I'm sorry, I misread. Your initial version was good, having a lifetime in the error will complicate things way too much. :)